cpp/modmesh/inout/gmsh.hpp

```diff
@@ -0,0 +1,320 @@
+#pragma once
+#include <string>
+#include <sstream>
+#include <vector>
+#include <unordered_map>
+#include <functional>
+
+#include <modmesh/base.hpp>
+#include <modmesh/mesh/mesh.hpp>
+#include <modmesh/buffer/buffer.hpp>
+
+namespace modmesh
+{
+namespace inout
+{
+struct GmshElementDef
+{
+    static GmshElementDef by_id(uint16_t id);
+
+    GmshElementDef(uint8_t ndim, uint16_t nnds, uint8_t mmtpn, small_vector<uint8_t> const & mmcl)
+        : m_ndim(ndim)
+        , m_nnds(nnds)
+        , m_mmtpn(mmtpn)
+        , m_mmcl(mmcl)
+    {
+    }
+
+    ~GmshElementDef() = default;
+
+    GmshElementDef() = default;
+    GmshElementDef(const GmshElementDef & other) = delete;
+    GmshElementDef(GmshElementDef && other) = delete;
+    GmshElementDef & operator=(const GmshElementDef & other) = delete;
+    GmshElementDef & operator=(GmshElementDef && other) = delete;
+
+    uint8_t ndim() const { return m_ndim; }
+    uint16_t nnds() const { return m_nnds; }
+    uint8_t mmtpn() const { return m_mmtpn; }
+    small_vector<uint8_t> mmcl() const { return m_mmcl; }
+
+private:
+    uint8_t m_ndim = 0; /* Number of dimension */
+    uint16_t m_nnds = 0; /* Number of nodes     */
+    uint8_t m_mmtpn = 0; /* modmesh cell type   */
+    small_vector<uint8_t> m_mmcl; /* modmesh cell order  */
+}; /* end struct GmshElementDef */
+
+class Gmsh
+    : public NumberBase<int32_t, double>
+{
+    using number_base = NumberBase<int32_t, double>;
+    using int_type = typename number_base::int_type;
+    using uint_type = typename number_base::uint_type;
+    using real_type = typename number_base::real_type;
+
+public:
+    explicit Gmsh(const std::string & data);
+
+    ~Gmsh() = default;
+
+    Gmsh() = delete;
+    Gmsh(Gmsh const & other) = delete;
+    Gmsh(Gmsh && other) = delete;
+    Gmsh & operator=(Gmsh const & other) = delete;
+    Gmsh & operator=(Gmsh && other) = delete;
+
+    std::shared_ptr<StaticMesh> toblock(void);
+
+private:
+    enum class FormatState
+    {
+        BEGIN,
+        META_END,
+        PYHSICAL_NAME_END,
+        NODE_END,
+        ELEMENT_END
+    };
+
+    // Check the finite state machine state transition is valid or not to check msh file format is correct
+    bool isValidTransition(const std::string s)
```

Comment:
Ah, I forgot to remind that C++ function names should follow snake_case.  You used snake_case in other functions.  This looks like an overlook.  Could you change it and review all in your PR?

Reply:
I fixed this in https://github.com/solvcon/modmesh/pull/233/commits/f7de14ff4eb817cc2c858ee14dfcaa4ec52f2a49
I changed `isValidTransition` and `toblock` to snake case in `Gmsh` class